pkg/cloudsql/queries.go

```diff
@@ -0,0 +1,58 @@
+package cloudsql
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"time"
+
+	monitoring "cloud.google.com/go/monitoring/apiv3/v2"
+	"cloud.google.com/go/monitoring/apiv3/v2/monitoringpb"
+	"google.golang.org/api/iterator"
+	"google.golang.org/protobuf/types/known/timestamppb"
+)
+
+func GetCPUUtilization(projectID string, databaseName string) (float64, error) {
+	return getLatestMetricValue(projectID, databaseName, "cloudsql.googleapis.com/database/cpu/utilization")
+}
+
+func getLatestMetricValue(projectID string, databaseName string, metricName string) (float64, error) {
+	ctx := context.Background()
+	databaseID := fmt.Sprintf("%s:%s", projectID, databaseName)
+
+	client, err := monitoring.NewMetricClient(ctx)
+	if err != nil {
+		return 0, err
+	}
+	defer client.Close()
+
+	endTime := time.Now().UTC()
+	// get the last _complete_ minute of data
+	startTime := endTime.Add(-10 * time.Minute)
+
+	filter := fmt.Sprintf(`metric.type = "%s" AND resource.label.database_id = "%s"`, metricName, databaseID)
+
+	req := &monitoringpb.ListTimeSeriesRequest{
+		Name:   fmt.Sprintf("projects/%s", projectID),
+		Filter: filter,
+		Interval: &monitoringpb.TimeInterval{
+			StartTime: &timestamppb.Timestamp{Seconds: startTime.Unix()},
+			EndTime:   &timestamppb.Timestamp{Seconds: endTime.Unix()},
+		},
+		View: monitoringpb.ListTimeSeriesRequest_FULL,
```

Comment:
So we get the full time-series but below it says `it.Next()` with the comment we only expect one result. Is there only one update per 10 minutes?

Reply:
Oh, so this is fun. This iterator is an iterator over possible time series that could get returned. i.e. if the filter wasn't specific enough then it would be possible to return multiple time series of a metric (e.g. one per databse if you had loads of databases) but I assumed here that we essentially want agent _per_ database so we just expect 1 metric time series

Reply:
Do we know if our `filter` is specific enough then? I'd rather error out in the case where we end up with multiple, as we might end up tracking the wrong database silently.

Reply:
So, our filters should be good enough, since the database name is always provided. It is just that the API accomodates getting many at the same time.

However, I can also assert that there is only one as you mention below